amd/src/management.js

```diff
@@ -0,0 +1,485 @@
+    deleteitems.forEach(element => {
+        element.addEventListener('click', async(e) => {
+            deleteModal(e, element.dataset.id, element.dataset.title, element.dataset.table);
+        });
+    });
+
+    // Add listener to preview items.
+    let previewitems = document.getElementsByClassName('preview');
+    previewitems.forEach(element => {
+        element.addEventListener('click', async(e) => {
+            previewModal(e);
+        });
+    });
+
+    // Add listener to select compcat to show corresponding items.
+    let compcats = document.getElementsByClassName('compcat');
+    compcats.forEach(element => {
+        element.addEventListener('click', async(e) => {
+            showItems(e, element.dataset.compcat);
+        });
+    });
+
+    // Add listener to manage component flavor relation.
+    let buttonicons = document.querySelectorAll('.buttonicons');
+    buttonicons.forEach(element => {
+        element.addEventListener('click', async(e) => {
+            compflavorModal(e);
+        });
+    });
+
+    let displaynamesbutton = document.getElementById('elements_displaynames_button');
+    displaynamesbutton.addEventListener('click', async(e) => {
+        displaynamesModal(e);
+    });
+
+    let displaynamesflavorbutton = document.getElementById('elements_displaynames_flavor_button');
+    displaynamesflavorbutton.addEventListener('click', async(e) => {
+        displaynamesFlavorModal(e);
+    });
+
+    let displaynamesvariantbutton = document.getElementById('elements_displaynames_variant_button');
+    displaynamesvariantbutton.addEventListener('click', async(e) => {
+        displaynamesVariantModal(e);
+    });
+
+    // Add listener to duplicate items.
+    let duplicateitems = document.getElementsByClassName('duplicate');
+    duplicateitems.forEach(element => {
+        element.addEventListener('click', async() => {
+            duplicateItem(element.dataset.id, element.dataset.table).always(() => reload());
+        });
+    });
+
+    // Add image and text to item setting click area.
+    let enlargeItems = document.querySelectorAll(
+        '.flavor .card-body > .clickingextended, .component .card-body > .clickingextended, .variant .card-body > .clickingextended'
+    );
+    enlargeItems.forEach(element => {
+        element.addEventListener('click', async(e) => {
+            let item = e.target.closest('.item');
+            item.querySelector('a.edit').click();
+        });
+    });
+
+    // After submitting a new item, reset active compcat.
+    if (params.compcatactive) {
+        let compcat = document.querySelector('.compcat[data-compcat="' + params.compcatactive + '"]');
+        if (compcat) {
+            showItems(false, params.compcatactive);
+            compcat.classList.add('active');
+        }
+    }
+};
+
+/**
+ * Show dynamic form to add/edit a source.
+ * @param {*} e
+ * @param {*} id
+ * @param {*} table
+ */
+function showModal(e, id, table) {
```

Comment:
stay in es6 notation?

```suggestion
showModal = (e, id, table) => {
```
Also check the other functions in all the files. Would be nice if we could use the same style.


what is `e`? Could you find a better name? At least `event` maybe? :)

Shouldn't this function be async and await the `show()`? Also double check the other functions that create modals.